# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/svg/SVGAnimatedViewBox.cpp
# Commit: b848ca90888e
# Full Hash: b848ca90888ee8e446f3d54b65c3ef68c71a1451
# Author: longsonr <longsonr@gmail.com>
# Date: 2019-06-20 22:06:31
# Regressor Bug: 1551030
# File Overlap Count: 2
# Description:
#   Bug 1551030 - Merge all SVGRect classes r=dholbert
# ==============================================================================

diff -r 20956a783e08 -r b848ca90888e dom/svg/SVGAnimatedViewBox.cpp
--- a/dom/svg/SVGAnimatedViewBox.cpp	Thu Jun 20 14:19:57 2019 +0300
+++ b/dom/svg/SVGAnimatedViewBox.cpp	Thu Jun 20 07:03:54 2019 -0700
@@ -9,6 +9,7 @@
 #include "mozilla/Move.h"
 #include "mozilla/SMILValue.h"
 #include "mozilla/SVGContentUtils.h"
+#include "mozilla/dom/SVGRect.h"
 #include "nsCharSeparatedTokenizer.h"
 #include "SVGViewBoxSMILType.h"
 #include "nsTextFormatter.h"
@@ -61,32 +62,9 @@
   return NS_OK;
 }
 
-/* Cycle collection macros for SVGAnimatedViewBox */
-
-NS_SVG_VAL_IMPL_CYCLE_COLLECTION_WRAPPERCACHED(SVGAnimatedViewBox::DOMBaseVal,
-                                               mSVGElement)
-NS_SVG_VAL_IMPL_CYCLE_COLLECTION_WRAPPERCACHED(SVGAnimatedViewBox::DOMAnimVal,
-                                               mSVGElement)
-
-NS_IMPL_CYCLE_COLLECTING_ADDREF(SVGAnimatedViewBox::DOMBaseVal)
-NS_IMPL_CYCLE_COLLECTING_RELEASE(SVGAnimatedViewBox::DOMBaseVal)
-
-NS_IMPL_CYCLE_COLLECTING_ADDREF(SVGAnimatedViewBox::DOMAnimVal)
-NS_IMPL_CYCLE_COLLECTING_RELEASE(SVGAnimatedViewBox::DOMAnimVal)
-
-NS_INTERFACE_MAP_BEGIN_CYCLE_COLLECTION(SVGAnimatedViewBox::DOMBaseVal)
-  NS_WRAPPERCACHE_INTERFACE_MAP_ENTRY
-  NS_INTERFACE_MAP_ENTRY(nsISupports)
-NS_INTERFACE_MAP_END
-
-NS_INTERFACE_MAP_BEGIN_CYCLE_COLLECTION(SVGAnimatedViewBox::DOMAnimVal)
-  NS_WRAPPERCACHE_INTERFACE_MAP_ENTRY
-  NS_INTERFACE_MAP_ENTRY(nsISupports)
-NS_INTERFACE_MAP_END
-
-static SVGAttrTearoffTable<SVGAnimatedViewBox, SVGAnimatedViewBox::DOMBaseVal>
+static SVGAttrTearoffTable<SVGAnimatedViewBox, SVGRect>
     sBaseSVGViewBoxTearoffTable;
-static SVGAttrTearoffTable<SVGAnimatedViewBox, SVGAnimatedViewBox::DOMAnimVal>
+static SVGAttrTearoffTable<SVGAnimatedViewBox, SVGRect>
     sAnimSVGViewBoxTearoffTable;
 SVGAttrTearoffTable<SVGAnimatedViewBox, SVGAnimatedRect>
     SVGAnimatedViewBox::sSVGAnimatedRectTearoffTable;
@@ -206,70 +184,45 @@
   return domAnimatedRect.forget();
 }
 
-already_AddRefed<SVGIRect> SVGAnimatedViewBox::ToDOMBaseVal(
+already_AddRefed<SVGRect> SVGAnimatedViewBox::ToDOMBaseVal(
     SVGElement* aSVGElement) {
   if (!mHasBaseVal || mBaseVal.none) {
     return nullptr;
   }
 
-  RefPtr<DOMBaseVal> domBaseVal = sBaseSVGViewBoxTearoffTable.GetTearoff(this);
+  RefPtr<SVGRect> domBaseVal = sBaseSVGViewBoxTearoffTable.GetTearoff(this);
   if (!domBaseVal) {
-    domBaseVal = new DOMBaseVal(this, aSVGElement);
+    domBaseVal = new SVGRect(this, aSVGElement, SVGRect::BaseValue);
     sBaseSVGViewBoxTearoffTable.AddTearoff(this, domBaseVal);
   }
 
   return domBaseVal.forget();
 }
 
-SVGAnimatedViewBox::DOMBaseVal::~DOMBaseVal() {
-  sBaseSVGViewBoxTearoffTable.RemoveTearoff(mVal);
+SVGRect::~SVGRect() {
+  if (mType == BaseValue) {
+    sBaseSVGViewBoxTearoffTable.RemoveTearoff(mVal);
+  } else if (mType == AnimValue) {
+    sAnimSVGViewBoxTearoffTable.RemoveTearoff(mVal);
+  }
 }
 
-already_AddRefed<SVGIRect> SVGAnimatedViewBox::ToDOMAnimVal(
+already_AddRefed<SVGRect> SVGAnimatedViewBox::ToDOMAnimVal(
     SVGElement* aSVGElement) {
   if ((mAnimVal && mAnimVal->none) ||
       (!mAnimVal && (!mHasBaseVal || mBaseVal.none))) {
     return nullptr;
   }
 
-  RefPtr<DOMAnimVal> domAnimVal = sAnimSVGViewBoxTearoffTable.GetTearoff(this);
+  RefPtr<SVGRect> domAnimVal = sAnimSVGViewBoxTearoffTable.GetTearoff(this);
   if (!domAnimVal) {
-    domAnimVal = new DOMAnimVal(this, aSVGElement);
+    domAnimVal = new SVGRect(this, aSVGElement, SVGRect::AnimValue);
     sAnimSVGViewBoxTearoffTable.AddTearoff(this, domAnimVal);
   }
 
   return domAnimVal.forget();
 }
 
-SVGAnimatedViewBox::DOMAnimVal::~DOMAnimVal() {
-  sAnimSVGViewBoxTearoffTable.RemoveTearoff(mVal);
-}
-
-void SVGAnimatedViewBox::DOMBaseVal::SetX(float aX, ErrorResult& aRv) {
-  SVGViewBox rect = mVal->GetBaseValue();
-  rect.x = aX;
-  mVal->SetBaseValue(rect, mSVGElement);
-}
-
-void SVGAnimatedViewBox::DOMBaseVal::SetY(float aY, ErrorResult& aRv) {
-  SVGViewBox rect = mVal->GetBaseValue();
-  rect.y = aY;
-  mVal->SetBaseValue(rect, mSVGElement);
-}
-
-void SVGAnimatedViewBox::DOMBaseVal::SetWidth(float aWidth, ErrorResult& aRv) {
-  SVGViewBox rect = mVal->GetBaseValue();
-  rect.width = aWidth;
-  mVal->SetBaseValue(rect, mSVGElement);
-}
-
-void SVGAnimatedViewBox::DOMBaseVal::SetHeight(float aHeight,
-                                               ErrorResult& aRv) {
-  SVGViewBox rect = mVal->GetBaseValue();
-  rect.height = aHeight;
-  mVal->SetBaseValue(rect, mSVGElement);
-}
-
 UniquePtr<SMILAttr> SVGAnimatedViewBox::ToSMILAttr(SVGElement* aSVGElement) {
   return MakeUnique<SMILViewBox>(this, aSVGElement);
 }